guides/use-microprofile-amqp-connector-with-ssl-openshift.adoc

```diff
@@ -0,0 +1,195 @@
+[source, bash, options="nowrap"]
+----
+amqp-host=<url-to-amq-broker-route>
+amqp-port=443
+amqp-username=admin
+amqp-password=admin
+amqp-use-ssl=true
+mp.messaging.connector.smallrye-amqp.wildfly.elytron.ssl.context=amqp-ssl-context
+
+mp.messaging.outgoing.source.connector=smallrye-amqp
+
+mp.messaging.incoming.in.connector=smallrye-amqp
+mp.messaging.incoming.in.address=source
+----
+
+[NOTE]
+--
+Update `amqp-host` - url of remote AMQ 7 broker deployed on Openshift, for example "amq-broker-all-0-svc-rte-mnovak.apps.mnovak-oegn.psi.domain.com". You can figure out this by checking `oc get routes` for `amq-broker-all-0-svc-rte` route.
+--
+
+`amqp-port` - port where AMQ 7 broker is listening, in case of openshift there is used port `443` for secured communication
+
+`amqp-username` and `amqp-password` - username and password for authentication to AMQ 7 broker. `admin/admin` is default used in AMQ 7 broker
+
+`amqp-use-ssl=true` - specifies that we want to use a secure connection when connecting to the broker.
+
+`mp.messaging.outgoing.source.connector=amqp-ssl-context` - this is not needed if there are CA signed certificates. However, in our case we're using self-signed certificates, so we need to specify a truststore
+in the Elytron subsystem and create an SSLContext referencing that. The value of this property is used to look up the SSLContext in the Elytron subsystem under /subsystem=elytron/client-ssl-context=* in the WildFly management model.
+In this case the property value is `amqp-ssl-context`, so we look up the SSLContext defined by /subsystem=elytron/client-ssl-context=amqp-ssl-context and use that to configure the truststore to use for the connection to AMQ broker.
+For this reason there is defined CLI script `scripts/config.cli` which will be used to configure client-ssl-context during build of trimmed Wildfly server:
+[source, ruby]
+----
+batch
+/subsystem=elytron/key-store=truststore-ssl-test:add(credential-reference={clear-text=changeit}, path=<path-to>/client.ts, type=PKCS12)
+/subsystem=elytron/trust-manager=truststore-ssl-test:add(key-store=truststore-ssl-test)
+/subsystem=elytron/key-store=keystore-ssl-test:add(credential-reference={clear-text=changeit}, path=<path-to>/client.ks, type=PKCS12)
+/subsystem=elytron/key-manager=keystore-ssl-test:add(key-store=keystore-ssl-test,credential-reference={clear-text=changeit})
+/subsystem=elytron/client-ssl-context=amqp-ssl-context:add(trust-manager=truststore-ssl-test,key-manager=keystore-ssl-test)
+run-batch
+----
+[NOTE]
+--
+Update `<path-to>` in above cli script to directory with generated truststore and keystore.
+--
+
+You can build the application by running:
+[source, bash]
+----
+mvn package
+----
+
+There is used Wildfly Maven Plugin to provision trimmed Wildfly application server with just the required modules. It deploys reactive messaging app into it and package all as bootable jar which can be started like:
+[source, bash]
+----
+java -jar ./target/server-bootable.jar
```

Comment:
so if you run your server locally, you should be able to run it with something like

```
AMQ_HOST=amq-broker-all-0-svc-rte-mnovak.apps.mnovak-oegn.psi.domain.com java -jar ./target/server-bootable.jar
```